fourcc.py

```diff
@@ -13,18 +13,10 @@
 mod_broadcom = re.compile(r'^#define (DRM_FORMAT_MOD_BROADCOM\w+(?=\s))', flags=re.M)
 
 with open(fourcc_include) as f:
-	for l in f.readlines():
-		ident = fmt.search(l)
-		if ident != None:
-			fmt_list.append(ident.group(1))
-			continue
-		ident = mod.search(l)
-		if ident != None:
-			mod_list.append(ident.group(1))
-			continue
-		ident = mod_broadcom.search(l)
-		if ident != None:
-			mod_list.append(ident.group(1))
+	str = f.read()
+	fmt_list = fmt.findall(str)
+	mod_list = mod.findall(str)
+	mod_list += set(mod_broadcom.findall(str)) - set(mod_list)
```

Comment:
I keep coming back with all of small comments, but this could become
```python
mod_list = set(mod.findall(str) + mod_broadcom.findall(str))
```
Order is not important at all.

Also, thinking about it, `str` is the name of a builtin function/type, so naming the variable this is overriding it. Damn dynamic languages...
We should change it to something else.

Reply:
I appreciate all of your comments. I must say though, despite order not mattering, what you suggest puts the mod formats out of order to the point where the generated file looks ugly. Having everything in the same order that is in the header file would make most sense to me but I digress. Formatted.

Idly I was wondering what you had in mind when you mentioned:
> Now that this is happening at build time, we could parse/generate all of the combinations for 
formats like this

where 'this' is the currently-special-cased `if ((modifier >> 56) == DRM_FORMAT_MOD_VENDOR_ARM)`
It seems like this case is the only one like it.

Reply:
>what you suggest puts the mod formats out of order to the point where the generated file looks ugly.

It would be nice to keep the generated code looking nice, but it's not really a concern. I mean, if you want to look at what modifiers there are, you'd just look in drm_fourcc.h.

>where 'this' is the currently-special-cased if ((modifier >> 56) == DRM_FORMAT_MOD_VENDOR_ARM)

Yeah, mainly that. Since the "argument" is a bitmask, you could generate cases of all possible combinations of bits.
That's effectively 2^11 (even if all combinations aren't technically valid), so that's **way** too much to do by hand, but for machine generated code, that's not a big deal.
`DRM_FORMAT_MOD_ARM_AFBC(BLOCK_SIZE_16x16, SPARSE|TILED|DB)`

But perhaps we just need to modify the function to not return a string literal so we can properly have extra format stuff added without generating huge lookup-tables.